src/main/java/com/dm/dmservicesandbox/domain/Role.java

```diff
@@ -0,0 +1,69 @@
+package com.dm.dmservicesandbox.domain;
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.Set;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+public class Role implements Serializable {
+    private static final long serialVersionUID = 1L;
+    public static final String USER = "USER";
+    public static final String ROLE_USER = "ROLE_USER";
+    public static final String ROLE_ADMIN = "ROLE_ADMIN";
+    public static final String ROLE_MODERATOR = "ROLE_MODERATOR";
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "ROLE_ID")
+    private Long roleId;
+
+    private String name;
+
+    // bi-directional many-to-many association to User
+    @ManyToMany(mappedBy = "roles")
```

Comment:
Should we keep a separate class to handle this relationship - UserRoleMapping and avoid  overridding equals/hashcode override here. I feel User has Roles and we should avoid having roles tracking which users are mapped. The many to many perhaps better to keep outside this class. thoughts?

Reply:
Role Class is doing the same

    // bi-directional many-to-many association to User
    @ManyToMany(mappedBy = "roles")
    private Set<UserLogin> users;